funding/manager_test.go

```diff
@@ -4429,8 +4449,36 @@ func TestFundingManagerZeroConf(t *testing.T) {
 		Tx: fundingTx,
 	}
 
+	// After 6 confirmation a channel which should be announced to the
+	// network will be readded to the network graph and announced by its
+	// real ShortChannelID. In case of a public zeroconf channel we need
+	// to make sure the MessageFlags of the prior edgeinfo are not copied
+	// but redetermined.
+	//nolint:lll
+	alice.fundingMgr.cfg.DeleteAliasEdge = func(scid lnwire.ShortChannelID) (*channeldb.ChannelEdgePolicy, error) {
+		//nolint:lll
+		return &channeldb.ChannelEdgePolicy{
+			MessageFlags:              lnwire.ChanUpdateRequiredMaxHtlc | lnwire.ChanUpdateDontForward,
+			FeeBaseMSat:               alice.fundingMgr.cfg.DefaultRoutingPolicy.BaseFee,
+			FeeProportionalMillionths: alice.fundingMgr.cfg.DefaultRoutingPolicy.FeeRate,
+			MaxHTLC:                   alice.fundingMgr.cfg.RequiredRemoteMaxValue(fundingAmt),
+			MinHTLC:                   bob.fundingMgr.cfg.DefaultMinHtlcIn,
+		}, nil
+	}
+	//nolint:lll
+	bob.fundingMgr.cfg.DeleteAliasEdge = func(scid lnwire.ShortChannelID) (*channeldb.ChannelEdgePolicy, error) {
```

Comment:
this causes a race condition because `waitForZeroConfChannel` tries to use the function

Reply:
I still kept the check after the public zeroconf channel passed the 6-conf target but moved the redefinition of the `DeleteAliasEdge` to the top where there can happen no race anymore.